### PR TITLE
Directed graph task roadmap

### PR DESCRIPTION
**Pull request by seibert**
Inspired by discussion with @stuartarchibald, this is some tooling to render tasks described in a YAML file as an interactive HTML DAG.

If this proves to be useful, we can plumb this into the sphinx rendering.

### CODE REVIEW

**Comment by sklam** [comment]
Some issues i noticed regarding the tooltip:
- when zooming the tooltip will get out-of-sync with the nodes and move out of view.  
- tooltip can hide other nodes.

My suggestion to fix this is to make a fixed pane for showing detail description.  Kind of like the google map layout.  The left pane contains the description so that it can have long details with independent scrolling.  The right larger pane will contain the DAG map.

**Comment by sklam** [comment]
Here's the template to implement my suggestion: https://gist.github.com/sklam/8ff3bfa55ad4f30f2ec7878161041f8e

**Comment by stuartarchibald** [comment]
Would be great to get this in for 0.46, was it just the couple of things above and some js/HTML tweaks left to do?

**Comment by seibert** [comment]
I think I've updated with suggestions, and also threw in some more testing tasks.

File: docs/dagmap/dagmap.yaml [hunk 1]
```diff
@@ -157,3 +157,31 @@ tasks:
...
+    description: |
+      Make it possible to use pytest as test runner for Numba
+
+  - label: Option to run modified tests only
+     id: run_new_tests
+    description: |
+      Use / make pytest plugin to detect all test files which are new / changed
+      relative to a given branch, and run only those tests
+    depends_on:
+	- pytest
```
Comment (stuartarchibald) [inline docs/dagmap/dagmap.yaml]: tabs?!

File: docs/dagmap/template.html [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+                    })
+                }
+            });
+        });
+    </script>
+
+
+</body>
+
+</html>
```
Comment (stuartarchibald) [inline docs/dagmap/template.html]: new line at EOF?

**Comment by seibert** [comment]
ok, had some words with my text editor (forgot that I didn't have yaml-mode on this system), and I think the indents and newlines are fixed now

File: docs/dagmap/README.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+conda install jinja2 python-graphviz pyyaml
+```
+
+## Usage
+
+```
+./render.py -o dagmap.html dagmap.yaml
+```
+
+The generated HTML file will look for `jquery.graphviz.svg.js` in the same directory.
```
Comment (stuartarchibald) [inline docs/dagmap/README.md]: long line

File: docs/dagmap/dagmap.yaml [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+      unnecessarily.
+    depends_on:
+      - declarative_overload
+      - improve_test_suite_tooling
+      - faster_pr_testing
+
+  - label: Unify and add more test suite tooling
+    id: improve_test_suite_tooling
+    description: |
+      Add tools to help with common patterns in testing and unify the ones we have, there's no need for 12 spellings of "is this Python 3"
```
Comment (stuartarchibald) [inline docs/dagmap/dagmap.yaml]: numerous long lines in here, if they are wrapped does it mess up the rendering ?

File: LICENSES.third-party [hunk 5]
```diff
@@ -146,6 +146,27 @@ See the License for the specific language governing permissions and
 limitations under the License.
 
 
+jquery.graphviz.svg (https://github.com/mountainstorm/jquery.graphviz.svg/)
```
Comment (stuartarchibald) [inline LICENSES.third-party]: For consistency with the above/below, should the title/prelude to the license inclusion etc say what its for and where it is in the code base?

File: docs/dagmap/README.md (hunk 3 above)
Comment (seibert) [reply docs/dagmap/README.md]: good catch

File: docs/dagmap/dagmap.yaml (hunk 4 above)
Comment (seibert) [reply docs/dagmap/dagmap.yaml]: It should be fine since I'm using the YAML multi-line string syntax.  Fixed.

File: LICENSES.third-party (hunk 5 above)
Comment (seibert) [reply LICENSES.third-party]: Good idea. Added an explanation and notes on the path where the JS is placed.

**Comment by stuartarchibald** [comment]
Looks good, thanks for the fixes.

**Pull request merged**